### PR TITLE
Added check if we need to remove old key

### PR DESCRIPTION
**Pull request by AlexeyShok**
Now the key is removed from the cache and reloaded only when needed

### CODE REVIEW

**Review by mingazhev — CHANGES_REQUESTED** [review]
When using double flow you will have no way to reset caches on all EDO instances.

I suggest another way:
1. Change double flow to memory flow
2. Reset the cache manually (remove a key) when needed (in controller method that updates agent info), with calling a special method

File: Api/Services/Agents/HttpBasedAgentContextService.cs [hunk 1]
```diff
@@ -40,8 +40,22 @@ public async ValueTask<Result<AgentContext>> GetAgentInfo()
...
-        public async Task RefreshAgentContext() =>
-            _currentAgentContext = await GetAgentContext();
+        public async Task RefreshAgentContext()
+        {
+            var identityClaim = _tokenInfoAccessor.GetIdentity();
+            var identityHash = identityClaim is not null
+                ? HashGenerator.ComputeSha256(identityClaim)
+                : string.Empty;
+
+            var key = _flow.BuildKey(nameof(HttpBasedAgentContextService), nameof(GetAgentInfo), identityHash);
```
Comment (mingazhev) [inline Api/Services/Agents/HttpBasedAgentContextService.cs:50]: Key generation must be extracted to method

Comment (mingazhev) [inline Api/Services/Agents/HttpBasedAgentContextService.cs:50]: Is double flow changed to memory flow?

File: Api/Services/Agents/HttpBasedAgentContextService.cs [hunk 2]
```diff
@@ -40,8 +40,22 @@ public async ValueTask<Result<AgentContext>> GetAgentInfo()
...
+            var identityClaim = _tokenInfoAccessor.GetIdentity();
+            var identityHash = identityClaim is not null
+                ? HashGenerator.ComputeSha256(identityClaim)
+                : string.Empty;
+
+            var key = GetKey(identityHash);
+
+            _flow.Remove(key);
+
+            _currentAgentContext = await _flow.GetOrSetAsync(
```
Comment (mingazhev) [inline Api/Services/Agents/HttpBasedAgentContextService.cs:54]: Memory flow is synchronous, not need to use async overload

**Review by mingazhev — APPROVED** [review]

**Review by kirillta — APPROVED** [review]

File: Api/Services/Agents/HttpBasedAgentContextService.cs (hunk 2 above)
Comment (AlexeyShok) [reply Api/Services/Agents/HttpBasedAgentContextService.cs:54]: Func in GetOrSetAsync is asynchronous, Func in GetOrSet is synchronous

**Pull request merged**